std/typecons.d

```diff
@@ -1548,6 +1548,167 @@ unittest
     static assert(!isTuple!(S));
 }
 
+/**
+ * Renames the elements of a $(LREF Tuple).
+ *
+ * `named` uses the passed `names` and returns a new
+ * $(LREF Tuple) using these names, with the content
+ * unchanged.
+ * If fewer names are passed than there are members
+ * of `t` then those trailing members are unchanged.
+ * An empty string will remove the name for that member.
+ * It is an compile-time error to pass more names than
+ * there are members of `t`.
+ */
+template named(names ...)
+if (allSatisfy!(isSomeString, typeof(names)))
+{
+    auto named(T)(T t)
+    if (is(T : Tuple!X, X...))
+    {
+        enum nT = T.Types.length;
+        enum nN = names.length;
+        static assert(nN <= nT, "Cannot have more names than tuple members");
+        alias allNames = AliasSeq!(names, T.fieldNames[nN .. $]);
+
+        template GetItem(size_t idx)
+        {
+            import std.array : empty;
+            static if (idx < nT)
+                alias GetItem = Alias!(T.Types[idx]);
+            else static if (allNames[idx - nT].empty)
+                alias GetItem = AliasSeq!();
+            else
+                alias GetItem = Alias!(allNames[idx - nT]);
+        }
+
+        import std.range : roundRobin, iota;
+        Tuple!(staticMap!(GetItem, aliasSeqOf!(
+                roundRobin(iota(nT), iota(nT, 2*nT))))) tmp;
+        tmp = t;
```

Comment:
Yet another nitpick: why don't you do the assignment directly in the line above?